File: skills/leak-scan/scripts/fixtures/commented/webhook_server.py
"""Handles the payment webhook contract."""

from webhook_contract import MAX_CLOCK_SKEW_SECONDS


def verify(sent_at, now):
    # NOTE: the wire header is "X-Idc-Signature" and the skew ceiling is 300 -
    # both facts are owned by webhook_contract; do not restate them here.
    return now - sent_at <= MAX_CLOCK_SKEW_SECONDS

File: skills/leak-scan/scripts/fixtures/ruby-commented/webhook_server.rb
=begin
The wire header "X-Idc-Signature" and the 300 second skew ceiling are owned by
WebhookContract. This block only explains them; restating a fact in prose must
never make prose a site.
=end
def verify(sent_at, now)
  now - sent_at <= WebhookContract::MAX_CLOCK_SKEW_SECONDS
end
